extensions/sql-database-projects/src/models/project.ts

```diff
@@ -73,14 +73,17 @@ export class Project {
 		}
 
 		// find all database references to include
-		for (let r = 0; r < this.projFileXmlDoc.documentElement.getElementsByTagName(constants.ArtifactReference).length; r++) {
-			if (this.projFileXmlDoc.documentElement.getElementsByTagName(constants.ArtifactReference)[r].getAttribute(constants.Condition) !== constants.NotNetCoreCondition) {
-				const filepath = this.projFileXmlDoc.documentElement.getElementsByTagName(constants.ArtifactReference)[r].getAttribute(constants.Include);
+		const references = this.projFileXmlDoc.documentElement.getElementsByTagName(constants.ArtifactReference);
+		for (let r = 0; r < references.length; r++) {
+			if (references[r].getAttribute(constants.Condition) !== constants.NotNetCoreCondition) {
+				const filepath = references[r].getAttribute(constants.Include);
 				if (!filepath) {
 					throw new Error(constants.invalidDatabaseReference);
 				}
 
-				this.databaseReferences.push(path.parse(filepath).name);
+				let nameNodes = references[r].getElementsByTagName(constants.DatabaseVariableLiteralValue);
```

Comment:
also just tried changing this in SSDT - looks like it could be any alias. In SSDT it doesnt seem to impact the tree

Reply:
Yup the tree displays the dacpac file name and checking if the reference already exists is done by comparing the file paths. This is just keeping track of the DatabaseVariableLiteralValue in case it's needed to write back to the sqlproj later.